File: src/Tickets/Ticket.java
package Tickets;

import Employee.Employees;

public abstract class Ticket {
    private static int serialNum = 0;
    private int uniqueNum;
    private Employees creator,slover;
    private boolean status;
    private String description,solution;

    public Ticket(Employees creator, String description) {
        serialNum++;
        this.solution = "";
        this.slover = null;
        this.status = true;
        this.creator = creator;
        this.description = description;
        setUniqueNum(serialNum);
    }




    public String getSolution() {
        return solution;
    }

    public void setSolution(String solution) {
        this.solution = solution;
    }

    public int getSerialNum() {
        return serialNum;
    }


    public Employees getCreator() {
        return creator;
    }

    public int getUniqueNum() {
        return uniqueNum;
    }

    public void setUniqueNum(int uniqueNum) {
        this.uniqueNum = uniqueNum;
    }

    public void setCreator(Employees creator) {
        this.creator = creator;
    }

    public Employees getSlover() {
        return slover;
    }

    public void setSlover(Employees slover) {
        this.slover = slover;
    }

    public boolean isStatus() {
        return status;
    }

    public void setStatus(boolean status) {
        this.status = status;
    }

    public String getDescription() {
        return description;
    }

    public void setDescription(String description) {
        this.description = description;
    }

}

File: Ticket.cpp
//
// Created by xurui on 2020/1/29.
//

#include <iostream>
#include "Ticket.h"

Ticket::Ticket(const std::string &descriptions, const std::shared_ptr<Employee> &creator): descriptions(descriptions),creator(creator){
status = true;
}

unsigned int Ticket::getId() const {
    return id;
}

void Ticket::setId(unsigned int id) {
    Ticket::id = id;
}

const std::string &Ticket::getDescriptions() const {
    return descriptions;
}

void Ticket::setDescriptions(const std::string &descriptions) {
    Ticket::descriptions = descriptions;
}

const std::shared_ptr<Employee> &Ticket::getCreator() const {
    return creator;
}

void Ticket::setCreator(const std::shared_ptr<Employee> &creator) {
    Ticket::creator = creator;
}

const std::string &Ticket::getSolution() const {
    return solution;
}

void Ticket::setSolution(const std::string &solution) {
    Ticket::solution = solution;
}

const std::shared_ptr<Employee> &Ticket::getSlover() const {
    return slover;
}

void Ticket::setSlover(const std::shared_ptr<Employee> &slover) {
    Ticket::slover = slover;
}

bool Ticket::isStatus() const {
    return status;
}

void Ticket::setStatus(bool status) {
    Ticket::status = status;
}

Ticket::~Ticket() {

}











